File: api/schemas/lldtech.py
from pydantic import BaseModel, UUID4
from typing import Optional
from datetime import datetime

class LLDTechBase(BaseModel):
    id: UUID4
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str
    created_at: datetime

    class Config:
        from_attributes = True

class LLDTechCreate(BaseModel):
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str

    class Config:
        from_attributes = True

class LLDTechUpdate(BaseModel):
    original_name: Optional[str] = None
    

    class Config:
        from_attributes = True

File: client/schema/lldtech.ts
export interface LLDTech {
    id: string;
    project_id: string;
    original_name: string;
    stored_name: string;
    file_path: string;
    created_at: Date;
}
    

export interface LLDTechCreate {
    project_id: string;
    original_name: string;
    stored_name: string;
    file_path: string;
}

export interface LLDTechUpdate {
    original_name: string;
}